pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -198,3 +172,27 @@ func buildRouteDestination() []networking.HTTPRouteDestination {
 		},
 	}
 }
+
+// getRouteWeights generates a list of integer percentages for route weights that sum to 100.
+// If the number of routes does not evenly divide 100, the weights are calculated as follows:
+// Round all the weights down, find the difference between that sum and 100, then distribute
+// the difference among the weights.
+func getRouteWeights(numRoutes int) []int {
```

Comment:
I'd like to see a test or at least two examples on this function to help visualize it.

Reply:
added test and example for this and more details in the comment